src/Sylius/Bundle/ApiBundle/Validator/Constraints/OrderShippingMethodAvailableValidator.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Validator\Constraints;
+
+use Sylius\Bundle\ApiBundle\Command\OrderTokenValueAwareInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\ShippingMethodInterface;
+use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Webmozart\Assert\Assert;
+
+final class OrderShippingMethodAvailableValidator extends ConstraintValidator
```

Comment:
I wonder whether we should not check this as an eligibility checker for the shipping, it makes more sense to me to use this concept. And what is more, it could be used in both contexts, UI and API without necessary of adding duplicated constraints and validators

Reply:
I moved the logic to the Eligibility validator.